### PR TITLE
Add Number of Training Examples Per Intent

### PR DESCRIPTION
**Pull request by ariella879**
**Proposed changes**:
- ...
Link to Issue: https://github.com/RasaHQ/rasa/issues/6134
I logged the number of examples per intent along with the total number of intents when training the nlu. 
**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by sara-tagger** [comment]
Thanks for submitting a pull request 🚀 @wochinge will take a look at it as soon as possible ✨

File: rasa/nlu/training_data/training_data.py [hunk 1]
```diff
@@ -476,10 +476,18 @@ def _split(_examples: List[Message], _count: int) -> None:
         return test, train
 
     def print_stats(self) -> None:
+        number_of_examples_for_each_intent = []
+        for key, value in self.number_of_examples_per_intent.items():
```
Comment (wochinge) [inline rasa/nlu/training_data/training_data.py:480]: A tiny bit more explicit:
```suggestion
        for intent_name, example_count in self.number_of_examples_per_intent.items():
```

File: rasa/nlu/training_data/training_data.py [hunk 2]
```diff
@@ -476,10 +476,18 @@ def _split(_examples: List[Message], _count: int) -> None:
...
+            number_of_examples_for_each_intent.append(f"intent: {key}, training examples: {value}   ")
+        newline = "\n"        
+        
         logger.info("Training data stats:")
         logger.info(
             f"Number of intent examples: {len(self.intent_examples)} "
             f"({len(self.intents)} distinct intents)"
+            '\n'
+            
+            f"{newline.join(number_of_examples_for_each_intent)}"
```
Comment (wochinge) [inline rasa/nlu/training_data/training_data.py:490]:
```suggestion
            f"\n{newline.join(number_of_examples_for_each_intent)}"        
```

**Comment by ariella879** [comment]
Thank you @wochinge! @chkoss I have made the requested changes to my branch including changing it to debug. Thank you for reviewing my pull request.

**Comment by ariella879** [comment]
@wochinge certainly! I have added it.

File: changelog/6134.improvement.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+I log the number of examples per intent, with an attempt at making the output somewhat easy to read.
```
Comment (tmbo) [inline changelog/6134.improvement.rst:1]:
```suggestion
Log the number of examples per intent during training. Logging can be enabled using ``rasa train --debug``.
```

File: rasa/nlu/training_data/training_data.py [hunk 4]
```diff
@@ -476,11 +476,19 @@ def _split(_examples: List[Message], _count: int) -> None:
         return test, train
 
     def print_stats(self) -> None:
+        number_of_examples_for_each_intent = []
+        for key, value in self.number_of_examples_per_intent.items():
+            number_of_examples_for_each_intent.append(f"intent: {key}, training examples: {value}   ")
```
Comment (tmbo) [inline rasa/nlu/training_data/training_data.py:481]:
```suggestion
            number_of_examples_for_each_intent.append(f"intent: {intent_name}, training examples: {example_count}   ")
```

**Comment by chkoss** [comment]
@ariella879 Thanks for including the changes!
One thing that's still needed is to reformat the files using `black` (see [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions). Currently, the build fails because of unnecessary white space. Could you do this please? After that your PR can be merged.

**Comment by ariella879** [comment]
@chkoss Thank you for letting me know! I have formatted it using black.

**Pull request merged**